lib/src/block_syntaxes/code_block_syntax.dart

```diff
@@ -59,4 +55,23 @@ class CodeBlockSyntax extends BlockSyntax {
 
     return Element('pre', [Element.text('code', content)]);
   }
+
+  bool _shouldEnd(BlockParser parser) {
+    var i = 1;
+    while (true) {
+      final nextLine = parser.peek(i);
+      if (nextLine == null) {
```

Comment:
Can you add a comment here on what this means, (I assume it means EOF, but I'm not sure).